### ISSUE TITLE
[Bug?] Voting on comments causes them to jump around due to ranking.

### ISSUE THREAD

**Issue by mister-monster** [LemmyNet/lemmy#89]

**Comment by dessalines** [comment]
This is a tough one actually, and I might need someone with more expertise to chime in on this.

I made the commenting experience to be live updating, both for votes, and new comments. So its resorting the existing comments every time someone votes, or a new comment comes in. This means that there's **no reason to refresh the page**, everyone is seeing the **same thing in real time**, and the hot comments will stay at the top for as long as your connection is up.

Of course the down-side of this is that scrolling is difficult... what if you're actively writing a reply, and that comment gets downvoted and moved around the page? Or if someone posts a wall of text, what happens then? 

I def don't want to make the user refresh the page, but I also do see the scrolling around as an issue. There's probably some way to "center" the page based on what you're reading, but I'm not sure how to do it.

**Comment by BaptisteGelez** [comment]
I don't know how this is currently implemented, but maybe at least to make the jumps more pleasant and less confusing, having a smooth animation of the comments moving/appearing could help.

Otherwise I think it is quite hard to solve this issue on your side (I know it is a recurrent issue in Mastodon for instance, and that the code that tries to handle it is full of "magic" code). However [browsers are making some efforts to avoid this kind of annoyance](https://hacks.mozilla.org/2019/03/scroll-anchoring-in-firefox-66/).

So you can either try to see what Mastodon does (I can try to find where it is handled if you want) for instance, or wait for browsers to behave better in these situations…

**Comment by pojntfx** [comment]
Another option (although, to be honest I don't know how I'd implement that) could be to fix the view of the user on the comment they are currently watching; this way, the comments might update and they would notice the scrollbar changing, but the UI wouldn't  "jump around".

**Comment by mister-monster** [comment]
Would it be difficult to create an exception for live updates i.e. votes don't update live? That way ranking won't change but everything else updates real time.

**Comment by dessalines** [comment]
This also isn't just an issue with voting, but for new comments, comment edits, comment deletes too... all can push the content you're viewing up or down the page.

**Comment by pyjammas** [comment]
I'm a programmer but I've worked with a number of awesome UX'ers. They unequivocally argue against in-place updates because these have lots of potential unintended consequences. Instead, if you're going to be real-time, do this in a less disruptive way.

The one rule they instilled in me is to never mess with the 'flow' of the document, the vertical one being primary.

So I'd say much as it is cool to update a comment's position based on upvotes, for one, avoid the desire to do so. Make it something indicated in the metadata at worst, but don't mess with a user's comment order without them desiring it.

**Comment by dessalines** [comment]
This isn't just an issue with vote resorting, but with new comments too. If you're viewing a comment, and a new one comes in, it would potentially push yours down. I'm not sure how exactly to "center" the scroll around the current content you're viewing.

**Comment by physkets** [comment]
> Live updating comments are a **solution** to reddit's problem of having to constantly refresh the page to see new content. Its easily the most annoying thing about the site.

I don't think that is a problem at all. You could make a poll, and I'm guessing people will agree.

> A lot of the reason people are moving to discord, twitter, masto, etc from reddit, is because they're auto-updating

They are completely different applications. People go to each of those things for different things.
